app/subapps/browser/browser.js

```diff
@@ -232,26 +230,40 @@ YUI.add('subapp-browser', function(Y) {
             }.bind(this));
       } else {
         // If it's not a bundle then it's a charm.
-        store.charm(entityId.replace('cs:', ''), {
-          'success': function(data) {
-            var charm = data.charm,
-                config = {};
-            Object.keys(charm.options).forEach(function(key) {
-              config[key] = charm.options[key]['default'];
-            });
-            // We call the env deploy method directly because we don't want the
-            // ghost inspector to open.
-            this.get('env').deploy(
-                charm.id,
-                charm.name,
-                config,
-                undefined, //config file content
-                1, // number of units
-                {}, //constraints
-                null); // toMachine
-            this.fire('autoplaceAndCommitAll');
-          }.bind(this)
-        });
+        charmstore.getEntity(
+            entityId.replace('cs:', ''),
+            function(charm) {
+              charm = charm[0];
+              var config = {},
+                  options = charm.get('options');
+              Object.keys(options).forEach(function(key) {
+                config[key] = options[key]['default'];
+              });
+              // We call the env deploy method directly because we don't want
+              // the ghost inspector to open.
+              this.get('env').deploy(
+                  charm.get('id'),
+                  charm.get('name'),
+                  config,
+                  undefined, //config file content
+                  1, // number of units
+                  {}, //constraints
+                  null); // toMachine
+              this.fire('autoplaceAndCommitAll');
+            }.bind(this),
+            function(error) {
+              var message = 'Unable to deploy target: ' + entityId;
+              try {
+                message = JSON.parse(error.currentTarget.responseText).Message;
```

Comment:
Non-actionable - We've been using Y.JSON in the past; should we just default to the browser's parser in the future?